### PR TITLE
[Routing] Add installation and minimal example to README

### PR DESCRIPTION
**Pull request by wouterj**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | n/a
| License       | MIT
| Doc PR        | symfony/symfony-docs#13431

Similair to what I did in #35552, this PR updates the README of the Routing component to include a minimal example and installation command.

### CODE REVIEW

File: src/Symfony/Component/Routing/README.md [hunk 1]
```diff
@@ -3,10 +3,48 @@ Routing Component
...
+use Symfony\Component\Routing\Matcher\UrlMatcher;
+use Symfony\Component\Routing\RequestContext;
+use Symfony\Component\Routing\Route;
+use Symfony\Component\Routing\RouteCollection;
+
+$route = new Route('/blog/{slug}', ['_controller' => BlogController::class]);
+$routes = new RouteCollection();
+$routes->add('blog_show', $route);
+
+$context = new RequestContext('/');
```
Comment (Tobion) [inline src/Symfony/Component/Routing/README.md]: the base url param `/` does not seem necessary and could just confuse people. I'd suggest to remove it.

Comment (wouterj) [reply src/Symfony/Component/Routing/README.md]: Thanks, I've removed it :)

**Comment by nicolas-grekas** [comment]
I was wondering about `CompiledUrlMatcher`. It's mentioned on the `create_framework/routing.rst`, but https://symfony.com/doc/current/routing.html doesn't link to it, isn't it? Is this something that needs to be improved?

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
I think this is good enough.

**Comment by fabpot** [comment]
Thank you @wouterj.

**Pull request merged**